lib/matplotlib/axes/_axes.py

```diff
@@ -1122,15 +1122,17 @@ def hlines(self, y, xmin, xmax, colors='k', linestyles='solid',
         if not np.iterable(xmax):
             xmax = [xmax]
 
-        y, xmin, xmax = cbook.delete_masked_points(y, xmin, xmax)
-
+        y, xmin, xmax = cbook._combine_masks(y, xmin, xmax)
         y = np.ravel(y)
-        xmin = np.resize(xmin, y.shape)
-        xmax = np.resize(xmax, y.shape)
-
-        verts = [((thisxmin, thisy), (thisxmax, thisy))
-                 for thisxmin, thisxmax, thisy in zip(xmin, xmax, y)]
-        lines = mcoll.LineCollection(verts, colors=colors,
+        xmin = np.ma.resize(xmin, y.shape)
+        xmax = np.ma.resize(xmax, y.shape)
+
+        # Keep all the invalid points instead of deleting them,
+        # convert verts to MaskedArry.
+        masked_verts = [np.ma.array(
+            [y_xmin, y_xmax])for y_xmin, y_xmax in zip(
+                        np.ma.array([xmin, y]).T, np.ma.array([xmax, y]).T)]
```

Comment:
The wrapping is really weird here. I also don't understand the comment, as the previous code didn't delete any points either, the `cbook.delete_masked_points` did.